File: 2.0/java/webhooks/createWebhookSmsChat.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.common.model.ResourceId;
import com.callfire.api.client.api.webhooks.model.ResourceType;
import com.callfire.api.client.api.webhooks.model.Webhook;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api login", "api password");
        Webhook webhook = new Webhook();
        webhook.setName("inbound sms webhook");
        webhook.setResource(ResourceType.INBOUND_TEXT);
        webhook.getEvents().add(ResourceType.ResourceEvent.FINISHED);
        webhook.setCallback("https://callback-service.com/listener");
        ResourceId id = client.webhooksApi().create(webhook);
    }
}

File: 2.0/csharp/webhooks/createWebhookSmsChat.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.Common.Model;
using CallfireApiClient.Api.Webhooks.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var webhook = new Webhook
        {
            Name = "inbound sms webhook",
            Resource = ResourceType.INBOUND_TEXT,
            Events = new HashSet<ResourceEvent> {ResourceEvent.FINISHED},
            Callback = "https://callback-service.com/listener"
        };
        ResourceId resource = client.WebhooksApi.Create(webhook);
    }
}
